File: front_end/src/services/productService.js
import axios from "axios";

const PRODUCT_URL = "http://localhost:8080/api/products"
export const getPageProduct = async (page, productTypeName, gtePrice, ltePrice, keyword) => {
    try {
        return await axios.get(PRODUCT_URL + `?page=${page}&productTypeName=${productTypeName}&gtePrice=${gtePrice}&ltePrice=${ltePrice}&keyword=${keyword}`)
    } catch (e) {
        return undefined;
    }
}
export const getProductByProductCode = async (productCode) => {
    try {
        const res = await axios.get(PRODUCT_URL + `/details?productCode=${productCode}`);
        return res.data;
    } catch (e) {
        return undefined;
    }
}
export const getListProductType = async () => {
    try {
        return await axios.get(PRODUCT_URL + `/product-types`);
    } catch (e){
        return undefined;
    }
}

File: back_end/src/main/java/com/example/back_end/service/impl/ProductService.java
package com.example.back_end.service.impl;

import com.example.back_end.dto.productDto.IProductDto;
import com.example.back_end.repository.IProductRepository;
import com.example.back_end.service.IProductService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.stereotype.Service;

@Service
public class ProductService implements IProductService {
    @Autowired
    private IProductRepository productRepository;

    @Override
    public Page<IProductDto> getPageProduct(Pageable pageable, String productTypeName, Double gtePrice, Double ltePrice, String keyword) {
        try {
            return productRepository.getPageProduct(pageable, "%" + productTypeName + "%", gtePrice, ltePrice, "%" + keyword + "%");
        } catch (Exception e) {
            return null;
        }
    }

    @Override
    public IProductDto getProductByProductCode(String productCode) {
        try {
            return productRepository.getProductByProductCode(productCode);
        } catch (Exception e) {
            return null;
        }
    }
}
